waku/node/waku_node.nim

```diff
@@ -193,6 +195,125 @@ proc connectToNodes*(
   # NOTE Connects to the node without a give protocol, which automatically creates streams for relay
   await peer_manager.connectToNodes(node.peerManager, nodes, source = source)
 
+## Waku Sync
+
+proc mountWakuSync*(
+    node: WakuNode,
+    maxFrameSize: int = DefaultMaxFrameSize,
+    syncInterval: timer.Duration = DefaultSyncInterval,
+    relayJitter: Duration = DefaultGossipSubJitter,
+    enablePruning: bool = true, # For testing purposes
+): Result[void, string] =
+  if not node.wakuSync.isNil():
+    return err("Waku sync already mounted, skipping")
+
```

Comment:
It seems to me that, if a node starts up with an existing `archive` we will not prime the `wakuSync` storage on startup with the items already existing in the Store. I imagine the behaviour would be that the sync protocol would believe these items do not exist in the Store and try to sync and store all these items from a neighbour? Or am I missing something?

Reply:
This is a valid point, if we had persistent negrnyropy storage then we wouldn't need to do this..but since we are using in memory storage..this has to be done.

Reply:
Ah yeah totally forgot about that.